contracts/liquidity_hub/pool-network/terraswap_router/src/contract.rs

```diff
@@ -369,6 +374,16 @@ fn simulate_swap_operations(
                     },
                 )?;
 
+                if max_spread.is_some() && res.return_amount > Uint128::zero() {
```

Comment:
So if the max_spread is passed and the return_amount is above zero then we can perform the spread assertion